### ISSUE TITLE
Add way to rotate text. Text box tool?

### ISSUE THREAD

**Issue by JulianGro** [xournalpp/xournalpp#1566]
**Is your feature request related to a problem? Please describe.**
There is no way to rotate text, as far as I can tell.

**Describe the solution you'd like**
Maybe add a text box tool, which has a little rotate icon in a corner, that you can move to rotate the text.

**Additional context**
A text box tool could also add more possibilities, like having a fixed size box, that does not increase in size when you write more text. This would make it possible to write text, without having to manually set newline characters.

**Comment by Technius** [comment]
Rotation will be implemented #918. We have considered adding a fixed size setting (and other settings such as color, text styles, etc.) to the text tool, but we'll have to experiment with the user experience first.

**Comment by rickhg12hs** [comment]
As a workaround, xournalpp's `Math TEX` can be used by adding `\RequirePackage{graphicx}` to `standalone.cfg` and then using something like
```
\text{\rotatebox[origin=c]{90}{My Rotated Text}}
```

**Comment by CsatiZoltan** [comment]
@rickhg12hs I don't have a *standalone.cfg* file in *$HOME/.xournalpp*. I created one, put `\RequirePackage{graphicx}` into it and restarted Xournalpp. No change,

```latex
\text{\rotatebox[origin=c]{90}{My Rotated Text}}
```
is still not valid. How did you achieve it?

**Comment by rickhg12hs** [comment]
@CsatiZoltan Sorry, I didn't give a full description.

Paraphrasing/simplifying:  `xournalpp` uses your already-installed `TeX` system to create its `Math TeX`.  Specifically, it uses the `TeX` [`standalone`](https://ctan.org/pkg/standalone) package/class and all of its dependencies.  You can configure the defaults, etc., for the `standalone` class by editing `standalone.cfg`.

Here's my current complete `standalone.cfg`:
```latex
$ cat ~/texmf/tex/latex/standalone/standalone.cfg 
\NeedsTeXFormat{LaTeX2e}
\ProvidesFile{standalone.cfg}[%
    2018/03/26
    v1.3a
    Default configuration file for 'standalone' class]%
%% Enabled the "varwidth" option if the "varwidth" package is available:
%%\IfFileExists{varwidth.sty}{%
%%    \standaloneconfig{varwidth}%
%%}{}%

%% Default options:
\standaloneconfig{crop}

%% Option which 'preview' should be loaded with
%% (will be ignored outside preamble)
\expandafter\ifx\csname @notprerr\endcsname\PassOptionsToPackage\else
\PassOptionsToPackage{active,tightpage}{preview}%
\fi

%% Enable 'preview' option by default:
%%\standaloneconfig{preview}

%% Remove the border:
\standaloneconfig{border=0pt}

%% Default preview border (used by standalone v0.x):
%%\standaloneconfig{border=0.50001bp}

%% Disable navigation symbols in beamer.
%% This must be done AtEndOfClass because the options are not processed yet,
%% so "beamer" mode is not enabled yet.
\AtEndOfClass{%
\ifstandalonebeamer
  \setbeamertemplate{navigation symbols}{}%
\fi
}
%% vim: ft=tex

\RequirePackage{xcolor}
\RequirePackage{graphicx}
```
All I did was copy my default [`TeX Live`](https://www.tug.org/texlive/) `my-texlive-install-dir/texmf-dist/tex/latex/standalone/standalone.cfg` to `~/texmf/tex/latex/standalone/standalone.cfg` and then add the last two lines. Now I can color and rotate my LaTeX text within xournalpp.  If you need more capability, you can `\RequirePackage{}` more packages with `standalone.cfg` and `xournalpp` will just use them.  `xournalpp` seems to pass whatever you enter in the `Math TeX` to `LaTeX`, so whatever valid `LaTeX` you enter there, `xournalpp` will show what gets rendered. E.g.,

```latex
\text{\rotatebox[origin=c]{90}{My Rotated Text}}
```

Hopefully `xournalpp` will add an arbitrary rotation capability for any object/region.  Until then, this is a way to rotate text.

**Comment by Rmano** [comment]
Hmm.... nice solution, but I was sure there was a way to add things to the LaTeX template for `xournalpp`, without having to touch the `standalone` configuration. Am I wrong? I seem not able to find it...

**Comment by rickhg12hs** [comment]
@Rmano IDK, I couldn't find it either.

**Comment by CsatiZoltan** [comment]
@rickhg12hs `~/texmf` did not exist so I created it. Thank you very much, now it works!

**Comment by rickhg12hs** [comment]
@CsatiZoltan Awesome!  Yeah, I didn't have `~/texmf` before wanting to rotate `xournalpp` text either.

**Comment by Rmano** [comment]
> @Rmano IDK, I couldn't find it either (other than modifying [source code](https://github.com/xournalpp/xournalpp/blob/99766c1a037a5060e498c8ff8808c52a4a916b7a/src/control/LatexController.cpp#L25)).

Hmm, this will be a nice feature request --- the possibility to add user-defined packages to the latex preamble. `xcolor` and `graphicx` anyway should be added by default I think (especially the second one), a lot of thing rely on the availability of `scalebox` and `rotatebox`).

**Comment by rickhg12hs** [comment]
@Rmano Looks like there is a fair amount of thought on this.  #1115 #1188

**Comment by CsatiZoltan** [comment]
Would be nice indeed. [Ipe](http://ipe.otfried.org/) supports it through the GUI.

![Ipe](https://user-images.githubusercontent.com/12742890/80361680-a5ada200-8881-11ea-8a1a-4dd68b41f64a.png)

**Comment by Rmano** [comment]
Yes, that would be nice, but even a stop-gap things like having (mock up):

![image](https://user-images.githubusercontent.com/6414907/80362388-db06bf80-8882-11ea-9fc9-2982feeaf6ad.png)

**Comment by rolandlo** [comment]
The LaTeX tool enhancements #1952 have now been merged, so if you update to the latest version, 
things like `\text{\rotatebox[origin=c]{40}{My Rotated Text}} `work if you add \RequirePackage{graphicx} to the global template file (see LaTeX panel in the settings).
